### PR TITLE
Add support for single-image batches + ReduceLROnPlateau 

### PR DESCRIPTION
**Pull request by nathancooperjones**
This PR sets out to fix two minor bugs in `octopod`. 

1. The current version of `octopod`'s **`learner.fit` does not support `torch.optim.lr_scheduler.ReduceLROnPlateau`**, since this particular learning rate scheduler must be passed with a `metric` to know when the learning rate has plateaued. I have added an `if/else` check to see if the scheduler is of type `torch.optim.lr_scheduler.ReduceLROnPlateau`, and, if so, step while passing in the validation loss, else keep the step call as normal. 
2. Although rare, sometimes the right combination of data and batch size will lead to a batch only containing a single data point. Currently, **the vision model fails to run a batch of size 1 due to a PyTorch issue with `nn.BatchNorm1d`** likely listed [here](torch.optim.lr_scheduler.ReduceLROnPlateau). 
My fix is two-fold: 1) ensure that, during training, a batch of size 1 is ignored, and 2) the vision model allows for a single image to be passed through the model for validation / inference. With this change, if a batch has a size of 1, it is silently thrown out, but I'm not sure if it's appropriate to add in a warning for that, since it is taken care of automatically now. 

Screen Shot 2020-07-14 at 1 12 46 PM 
^ yay! :) 

Pull Request Checklist
 - [x] All tests in the `tests` folder pass with a local build
 - [X] Pull request includes a description of why we are doing this
 - [X] Init files import new capabilities to appropriate level of package (if applicable)
 - [X] CHANGELOG has been updated
 - [X] Version in `_version.py` has been updated
 - [X] README has been updated (if applicable)
 - [X] PR made from branch into WIP or WIP into Master

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,8 +55,7 @@ nvidia-docker run \
     --rm \
     -v "${PWD}:/octopod" \
     -p 8888:8888 \
-    -p 8000:8000 \
```
Comment (nathancooperjones) [inline README.md:58]: Since we aren't running a Flask app, I don't think we will need this port open. Is that right?

Comment (sugi-chan) [reply README.md:58]: hmm 0 idea? that might be right? @mcundiff3 thoughts?

File: octopod/learner.py [hunk 2]
```diff
@@ -160,6 +161,12 @@ def fit(
                 pbar
             )
 
+            if not step_scheduler_on_batch:
```
Comment (sugi-chan) [inline octopod/learner.py:164]: ok so we just moved this down so it is farther past the evaluation and then pass the val loss to the reduce on plateau if it is the scheduler.... im wondering if there is a cleaner way to handle this that is more flexible, but i think this is ok.

File: octopod/vision/models/multi_task_resnet.py [hunk 3]
```diff
@@ -90,7 +90,11 @@ def forward(self, x):
         full_img = self.resnet(x['full_img']).squeeze()
         crop_img = self.resnet(x['crop_img']).squeeze()
 
-        full_crop_combined = torch.cat((full_img, crop_img), 1)
+        if x[next(iter(x))].shape[0] == 1:
```
Comment (sugi-chan) [inline octopod/vision/models/multi_task_resnet.py:93]: throwback to my old hacky stuff

File: octopod/learner.py (hunk 2 above)
Comment (nathancooperjones) [reply octopod/learner.py:164]: I'm thinking that moving the scheduler step until after the validation loss won't affect anything else, even using another scheduler, since the scheduler will only affect the next time we are training. Does that sound right?

File: octopod/vision/models/multi_task_resnet.py (hunk 3 above)
Comment (nathancooperjones) [reply octopod/vision/models/multi_task_resnet.py:93]: very _good_ hacky stuff!

**Comment by nathancooperjones** [comment]
I have some ideas on how to better organize how we use schedulers, but I think that might be a bigger change than this PR. Maybe for a future Nate to take a stab at ;)

File: README.md (hunk 1 above)
Comment (mcundiff3) [reply README.md:58]: Yeah  you probably only need 8888 but I also don't think it really hurts anything to open it when running. Trying to think if there's anything else that would use the 8000 port. You could try running a notebook on an EC2 instance without surfacing the port and make sure it works.

Comment (nathancooperjones) [reply README.md:58]: Just tested - able to open Jupyter and run all the tests with just port 8888 exposed. I think we're okay to not require port 8000 open as well!

File: octopod/learner.py (hunk 2 above)
Comment (mcundiff3) [reply octopod/learner.py:164]: Yeah I'd love if there was a way to do this without hardcoding ReduceLROnPlateau in there. I know you all chatted already so I'm not sure what was covered but maybe we can have a conversation about alternatives.
